tracer/src/Datadog.Trace/AppSec/Waf/NativeBindings/NativeLibrary.cs

```diff
@@ -70,9 +70,9 @@ internal static bool TryLoad(string libraryPath, out IntPtr handle)
 
         internal static bool CloseLibrary(IntPtr library)
         {
-            if (library != IntPtr.Zero)
+            if (library == IntPtr.Zero)
             {
-                Log.Error("Trying to close a library with a null pointer");
+                Log.Warning("Trying to close WAF library with a null pointer");
```

Comment:
I think this should stay as error tbh
```suggestion
                Log.Error("Trying to close WAF library with a null pointer");
```